core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -589,10 +594,10 @@ public void process(@Nullable MappingMetaData mappingMd, boolean allowIdGenerati
         }
 
         // generate id if not already provided and id generation is allowed
-        if (allowIdGeneration) {
-            if (id == null) {
-                id(UUIDs.base64UUID());
-            }
+        if (allowIdGeneration && id == null) {
+            assert autoGeneratedID == false;
+            autoGeneratedID = true;
```

Comment:
can we explicitly set can have duplicates to false here (and default it true)? or at least assert it is false? (I prefer the first)